File: 387.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        vector<int> dict(128, 0);
        
        for(auto c: s){
            dict[c]++;
        }
        
        for(int i=0;i<s.size();++i){
            if(dict[s[i]] == 1){
                return i;
            }
        }
        
        return -1;
    }
};

File: Python/387.py
class Solution(object):
    def firstUniqChar(self, s):
        """
        :type s: str
        :rtype: int
        """
        d = defaultdict(int)
        for c in s:
            d[c] +=1
        for i in range(len(s)):
            if d[s[i]] == 1:
                return i
        return -1
